Code/Framework/AzCore/AzCore/Console/Console.cpp

```diff
@@ -110,7 +110,21 @@ namespace AZ
         ConsoleFunctorFlags requiredClear
     )
     {
-        return DispatchCommand(command, commandArgs, silentMode, invokedFrom, requiredSet, requiredClear);
+        if (!DispatchCommand(command, commandArgs, silentMode, invokedFrom, requiredSet, requiredClear))
+        {
+            // If the command could not be dispatched at this time add it to the deferred commands queue
+            DeferredCommand deferredCommand{ AZStd::string_view{ command },
+                                             DeferredCommand::DeferredArguments{ commandArgs.begin(), commandArgs.end() },
+                                             silentMode,
+                                             invokedFrom,
+                                             requiredSet,
+                                             requiredClear };
+
+            m_deferredCommands.emplace_back(AZStd::move(deferredCommand));
+            return false;
```

Comment:
Nitpick: It might be better to change this to either return an enum (return PerformCommandResult::Defferred;) or an AZ::Outcome, to make it clear to callers that getting a non-success result doesn't mean their command won't eventually run.

Especially because right now, the comment on the interface says it returns true on success, and false otherwise. In this case, you don't know if it's successful or not on running the command yet because it's been deferred, so it would be useful for callers to know it's been deferred and not just failed to invoke.

https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/IConsole.h#L46

Reply:
True, I was thinking the same. This is a widely used method so it'll definitely require proper deprecation..

Reply:
If an `AZ::Outcome<void, AZStd::string>` is returned, it should be pretty transparent to the user.

Reply:
Doing a search, there is only one location in the codebase where we check the result of PerformCommand and that is an the SettingsRegistryConsoleUtilsTest.

https://sourcegraph.com/search?q=context:global+repo:github%5C.com/o3de/+PerformCommand&patternType=standard&sm=1&groupBy=repo

Reply:
Ah I see, AZ::Outcome has a bool operator. Wow, I'm surprised by the amount of times we use AZ::Outcome<void,.,.>Enum seems more natural... I think Outcome should be when there's a return value the user actually wants to retrieve. I'm cool using enum even if it requires the deprecating step if it makes more sense. Thoughts?

Reply:
Fixed!